models/reviews.js

```diff
@@ -1,19 +1,31 @@
 const db = require("../db/connection");
 
 exports.selectReview = (req) =>{
-    return db.query(`SELECT review_id, title, review_body, designer, review_img_url, votes, category, owner, created_at
+    return db.query(`
+    SELECT reviews.review_id, reviews.title, reviews.review_body, reviews.designer, reviews.review_img_url, reviews.votes, reviews.category, reviews.owner, reviews.created_at, COUNT(reviews.review_id)::INT AS comment_count
     FROM reviews
-    WHERE review_id = $1`, [req.params.review_id]).then((review) =>{
-        if(review.rows.length === 0){
+    JOIN comments ON reviews.review_id = comments.review_id
+    WHERE reviews.review_id = $1
+    GROUP BY reviews.review_id`, [req.params.review_id]).then((review) =>{
+        return Promise.all(
+            [db.query(`SELECT * FROM reviews where review_id = $1`, [req.params.review_id]), review])
```

Comment:
What's the use of the Promise.all here? It looks like you are trying to have one function to handle retrieving both multiple and single reviews. This is using one function to handle two separate behaviours and I would extract them out into two separate functions. Be careful of making functions too general, rendering them difficult to debug.